File: src/de/redgames/somebot/SomeoneMessage.java
package de.redgames.somebot;

import net.dv8tion.jda.core.EmbedBuilder;
import net.dv8tion.jda.core.OnlineStatus;
import net.dv8tion.jda.core.entities.Member;
import net.dv8tion.jda.core.entities.TextChannel;

import java.awt.*;
import java.util.List;
import java.util.Random;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class SomeoneMessage {
    public static void sendSomeoneMessage(TextChannel channel, Member sender) {
        Stream<Member> memberStream = channel.getMembers().stream();

        memberStream = memberStream.filter(s -> s.getOnlineStatus() != OnlineStatus.OFFLINE);
        memberStream = memberStream.filter(s -> !s.getUser().isBot());
        memberStream = memberStream.filter(s -> s != sender);

        List<Member> memberList = memberStream.collect(Collectors.toList());
        Random random = new Random();
        Member member = memberList.get(random.nextInt(memberList.size()));


        EmbedBuilder embedBuilder = new EmbedBuilder();
        embedBuilder.setAuthor(sender.getEffectiveName());
        embedBuilder.setDescription(channel.getJDA().getSelfUser().getAsMention() + " **" + generateSmiley(random) + "** " + member.getAsMention());
        embedBuilder.setColor(new Color(random.nextInt(256), random.nextInt(256), random.nextInt(256)));

        channel.sendMessage(embedBuilder.build()).queue();
    }

    private static String generateSmiley(Random random) {
        switch(random.nextInt(10)) {
            case(0): return "(∩ ͡° ͜ʖ ͡°)⊃━☆ﾟ. o ･ ｡ﾟ";
            case(1): return "༼ つ ◕_◕ ༽つ";
            case(2): return "ヽ༼ ಠ益ಠ ༽ﾉ";
            case(3): return "(◕‿◕✿)";
            case(4): return "¯(°_o)/¯";
            case(5): return "¯\\_(ツ)_/¯";
            case(6): return "（✿ ͡◕ ᴗ◕)つ━━✫・o。";
            case(7): return "ಠ_ಠ";
            case(8): return "(⁄ ⁄•⁄ω⁄•⁄ ⁄)";
            case(9): return "(╯°□°）╯︵ ┻━┻";
        }
        return "";
    }
}

File: src/main/kotlin/de/redgames/somebot/SomeoneMessage.kt
package de.redgames.somebot

import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.OnlineStatus
import net.dv8tion.jda.api.entities.Member
import net.dv8tion.jda.api.entities.TextChannel
import java.awt.Color
import java.util.*

object SomeoneMessage {
    private val smileys = arrayOf(
        "(∩ ͡° ͜ʖ ͡°)⊃━☆ﾟ. o ･ ｡ﾟ",
        "༼ つ ◕_◕ ༽つ",
        "ヽ༼ ಠ益ಠ ༽ﾉ",
        "(◕‿◕✿)",
        "¯(°_o)/¯",
        "¯\\_(ツ)_/¯",
        "（✿ ͡◕ ᴗ◕)つ━━✫・o。",
        "ಠ_ಠ",
        "(⁄ ⁄•⁄ω⁄•⁄ ⁄)",
        "(╯°□°）╯︵ ┻━┻"
    )

    private fun generateSmiley(random: Random) = smileys[random.nextInt(smileys.size)]

    fun send(channel: TextChannel, sender: Member) {
        val members = channel.members
            .filter { it.onlineStatus != OnlineStatus.OFFLINE }
            .filter { !it.user.isBot }
            .filter { it != sender }
            .toMutableList()

        println(members)

        val random = Random()
        val member = members[random.nextInt(members.size)]

        val embedBuilder = EmbedBuilder()
        embedBuilder.setAuthor(sender.effectiveName)
        embedBuilder.setDescription("${channel.jda.selfUser.asMention} **${generateSmiley(random)}** ${member.asMention}")
        embedBuilder.setColor(Color(random.nextInt(256), random.nextInt(256), random.nextInt(256)))

        channel.sendMessage(embedBuilder.build()).queue()
    }
}